internal/cli/target.go

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+package cli
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/erikgeiser/promptkit/selection"
+	"kraftkit.sh/unikraft/target"
+)
+
+// SelectTarget is a utility method used in a CLI context to prompt the user
+// for a specific application's target.
+func SelectTarget(targets target.Targets) (target.Target, error) {
+	names := make([]string, 0, len(targets))
+	for _, t := range targets {
+		names = append(names, fmt.Sprintf(
+			"%s (%s)",
+			t.Name(),
+			target.TargetPlatArchName(t),
+		))
+	}
+
+	sort.Strings(names)
+
+	sp := selection.New("select target:", names)
+	sp.Filter = nil
+
+	result, err := sp.RunPrompt()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, t := range targets {
+		if fmt.Sprintf("%s (%s)", t.Name(), target.TargetPlatArchName(t)) == result {
+			return t, nil
+		}
+	}
+
+	return nil, fmt.Errorf("could not select target")
+}
+
+// FilterTargets returns a subset of `targets` based in input strings `arch`,
+// `plat` and/or `targ`
+func FilterTargets(targets target.Targets, arch, plat, targ string) (target.Targets, error) {
+	var selected target.Targets
+
+	// Filter `targets` by input arguments `arch`, `plat` and/or `targ`
+	for _, t := range targets {
+		switch true {
+		case
+			// If no arguments are supplied
+			len(targ) == 0 &&
+				len(arch) == 0 &&
+				len(plat) == 0,
```

Comment:
💯 re: the switch func

For the empty check, [`go build` emits the same assembly for `s == ""` and `len(s) == 0`](https://godbolt.org/z/fib1x1)

Reply:
I know, I'm only referring to code style here.
If we insist on using `len(s) == 0` instead of comparisons to empty strings, we should be consistent about it.

Reply:
I just realized this code had been relocated from elsewhere. I should have just ignored it, sorry!

Reply:
Right now the project is mixed, we should decide on one vs. the other. Any preference? We can then enforce it and also open an issue which reflects changing all instances from one to the other.